modules/consensus/accept_bench_test.go

```diff
@@ -0,0 +1,44 @@
+package consensus
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+// benchmarkEmptyBlocks is a benchmark that mines many blocks, and
+// measures how long it takes to add them to the consensusset
+func benchmarkEmptyBlocks(bt *testing.B) error {
+	bt.StopTimer()
+
+	// The test dir will be reset each time the benchmark
+	// is done.
+	cst, err := createConsensusSetTester("BenchmarkEmptyBlocks")
+	if err != nil {
+		return errors.New("Error creating tester: " + err.Error())
+	}
+
+	for j := 0; j < bt.N; j++ {
+		b, _ := cst.miner.FindBlock()
+		bt.StartTimer()
+		err := cst.cs.AcceptBlock(b)
+		if err != nil {
+			errstr := fmt.Sprintf("Error accepting %d: %s", j, err.Error())
+			return errors.New(errstr)
+		}
+		cst.csUpdateWait()
+		bt.StopTimer()
```

Comment:
I don't think it makes sense to include the csUpdateWait() inside of the timing, because that's going to stall until all modules have received the update and processed it. You're timing the modules as well as the consensus package.